src/include/Context.h

```diff
@@ -75,7 +76,9 @@ class Context {
   virtual ~Context() {}       // we want a virtual destructor!!!
   virtual void complete(int r) {
     finish(r);
+#if !defined(WITH_SEASTAR) || (defined(WITH_SEASTAR) && !defined(WITH_BLUESTORE))
     delete this;
+#endif
```

Comment:
@liu-chunmei could you add a comment to address this comment. i also had trouble understanding this.

Reply:
this is outdated,  currently it is #ifndef WITH_SEASTAR. in AlienStore, I need keep callback function OnCommit alive because there is a promise in it to set value so alien store can get_future to invoke to do continuation.  if delete itself in callback function ( at here), alien store continuation can't woken up.  if I identify cyanstore and alienstore, need add namespace for these callback context, so I treat cyanstore and alienstore the same. delete it in store functions.